core/api/src/services/kratos/totp.ts

```diff
@@ -128,34 +128,15 @@ const refreshToken = async (authToken: AuthToken): Promise<void | KratosError> =
   }
 }
 
-export const kratosRemoveTotp = async (token: AuthToken) => {
-  let flow: string
-
-  const res = await refreshToken(token)
-  if (res instanceof Error) return res
-
-  try {
-    const res = await kratosPublic.createNativeSettingsFlow({ xSessionToken: token })
-    flow = res.data.id
-  } catch (err) {
-    return new UnknownKratosError(err)
-  }
-
+export const kratosRemoveTotp = async (userId: UserId) => {
   try {
-    await kratosPublic.updateSettingsFlow({
-      flow,
-      updateSettingsFlowBody: {
-        method: "totp",
-        totp_unlink: true,
-      },
-      xSessionToken: token,
+    const removeTotpResponse = await kratosAdmin.deleteIdentityCredentials({
```

Comment:
Had to use the admin API here because we need an elevated session (TOTP verified) token, but the temporary token we are creating is not TOTP elevated.